src/qililab/platform/platform.py

```diff
@@ -465,6 +469,66 @@ def get_parameter(self, alias: str, parameter: Parameter, channel_id: ChannelID
         element = self.get_element(alias=alias)
         return element.get_parameter(parameter=parameter, channel_id=channel_id)
 
+    def data_draw_oscilloscope(self):
+        """From the runcard retrieve the parameters necessary to draw the qprogram."""
+        data_osci = {}
+        # All the aliases of the platform in a list
+        alias = [a["alias"] for a in map(asdict, self.runcard.buses)]
+        buses = [self.buses.get(alias=a) for a in alias]
+        instruments = {
+            instrument for bus in buses for instrument in bus.instruments if isinstance(instrument, (QbloxModule))
+        }
+        if all(isinstance(instrument, QbloxModule) for instrument in instruments):
+            for bus in buses:
+                for instrument, _ in zip(bus.instruments, bus.channels):
+                    if isinstance(instrument, QbloxModule):
+                        data_osci[bus.alias] = {}
+                        static_offset_i = 0
+                        static_offset_q = 0
+                        param = [
+                            Parameter.IF,
+                            Parameter.GAIN_I,
+                            Parameter.GAIN_Q,
+                            Parameter.OFFSET_I,
+                            Parameter.OFFSET_Q,
+                            Parameter.HARDWARE_MODULATION,
+                        ]
+                        for p in param:
+                            val = self.get_parameter(bus.alias, p)
+                            data_osci[bus.alias][p] = val
```

Comment:
if you need the "string" representation of the parameter you can do `data_osci[bus.alias][p.value] = val` here. Then you won't need this copying loop at the end.